src/main/java/org/ambraproject/wombat/service/PeerReviewServiceImpl.java

```diff
@@ -134,11 +144,34 @@ String getAllReviewsAsXml(List<Map<String, ?>> reviewLetterItems) throws IOExcep
     // append the acceptance letter
     peerReviewContent += acceptanceLetter;
 
+    peerReviewContent += "<received-date>" + articleMetadata.get("receivedDate") + "</received-date>";
+
     // wrap it in a root node
     peerReviewContent = "<peer-review>" + peerReviewContent + "</peer-review>";
     return peerReviewContent;
   }
 
+   /**
+   * Get article metadata to include in peer review history, such as the
+   * received date (aka, the original submission date).
+   *
+   * @param itemTable
+   * @return map with article metadata. 
+   */
+  Map<String,String> getArticleMetadata(Map<String,?> itemTable) throws IOException {
+    Map<String,String> metadata = new HashMap<>();
+
+    for (Object itemObj : new TreeMap(itemTable).values()) {
+
+      if (((Map<String,?>) itemObj).get("itemType").equals("article")) {
+        String articleXml = getContentXml((Map<String,?>)itemObj, "manuscript");
```

Comment:
https://github.com/PLOS/wombat/blob/develop/src/main/java/org/ambraproject/wombat/service/ArticleServiceImpl.java#L77

Here's an example where we get the article.  You might want to do something like this.  It implies the expectation of only one article.  Then you could get the date, and fail early if the article was unexpectedly not found.

Reply:
Beautiful. I wanted to do something like this instead of brute-force iteration. have replaced with this.